crt_portal/cts_forms/models.py

```diff
@@ -687,9 +688,30 @@ class ReportDispositionBatch(models.Model):
     """A group of reports that have been disposed of together."""
     uuid = models.UUIDField(primary_key=True, default=uuid.uuid4, unique=True, editable=False)
     disposed_date = models.DateTimeField(auto_now_add=True)
+    create_date = models.DateTimeField(default=datetime.today())
+    proposed_disposal_date = models.DateTimeField(blank=True, null=True)
     disposed_by = models.ForeignKey(User, related_name="disposed_report_batches", on_delete=models.PROTECT)
     disposed_count = models.IntegerField(default=0)
 
+    def add_records_to_batch(self, queryset, user):
+        """Creates a batch of disposed reports."""
+        current_request = CrequestMiddleware.get_request()
+        if not current_request:
+            raise ValueError("Cannot determine the current user for report disposal.")
+
+        if not user:
+            raise ValueError("Cannot determine the current user for report disposal.")
```

Comment:
Nice guard clauses - since the cases are simple, this could be DRY'ed up:

```suggestion
        if not current_request or not user:
            raise ValueError("Cannot determine the current user for report disposal.")
```